### PR TITLE
Add StackBuilder to help us generate aligned stack offsets.

### PR DESCRIPTION
**Pull request by vext01**

### CODE REVIEW

File: ykcompile/src/stack_builder.rs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    /// Allocate an object of given size and alignment on the stack, returning an offset which
+    /// is intended to be subtracted from the base pointer.
+    pub fn alloc(&mut self, size: u64, align: u64) -> u64 {
+        self.stack_top = Self::align(self.stack_top, align);
+        self.stack_top += size;
+        self.stack_top
+    }
+
+    /// Aligns `offset` to `align` bytes.
+    fn align(offset: u64, align: u64) -> u64 {
```
Comment (ptersilie) [inline ykcompile/src/stack_builder.rs:23]: Should we make this non-static so we can just use `self.stack_top`?

**Comment by vext01** [comment]
Try this. I also amended a comment to make it more concise / correct.

**Comment by ptersilie** [comment]
Looks good. Please squash.

**Comment by vext01** [comment]
splat.

**Pull request merged**